Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -446,7 +446,7 @@ void MemoryWidget::OnSearchAddress()
   bool good_addr, good_offset;
   // Returns 0 if conversion fails
   const u32 addr = m_search_address->text().toUInt(&good_addr, 16);
-  const u32 offset = m_search_offset->text().toUInt(&good_offset, 16);
+  const s32 offset = m_search_offset->text().toInt(&good_offset, 16);
```

Comment:
note that without casts below, this introduces an implicit sign/unsigned mismatch in terms of arithmetic. Ditto for the other cases as well